packaging/test_plugin_unified_storage_tiering.py

```diff
@@ -776,6 +777,83 @@ def test_put_and_get(self):
+
+                    admin_session.assert_icommand('imeta rm -R ufs1 irods::storage_tiering::minimum_restage_tier true')
+                    admin_session.assert_icommand('imeta add -R ufs1 irods::storage_tiering::minimum_restage_tier false')
+
+                    lib.create_local_testfile(filename)
+                    admin_session.assert_icommand('iput -R ufs0 ' + filename)
+                    self.assertTrue(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs0'))
+
+                    # Tier out replica from ufs0 to ufs1.
+                    sleep(6)
+                    invoke_storage_tiering_rule()
+                    admin_session.assert_icommand('iqstat', 'STDOUT', 'irods_policy_storage_tiering')
+
+                    # Ensure that the replica was tiered out to ufs1 and not preserved on ufs0.
+                    delay_assert_icommand(admin_session, 'ils -L ' + filename, 'STDOUT_SINGLELINE', 'ufs1')
+                    self.assertFalse(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs0'))
+                    self.assertFalse(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs2'))
+
+                    # Tier out replica from ufs1 to ufs2.
+                    sleep(15)
+                    invoke_storage_tiering_rule()
+                    admin_session.assert_icommand('iqstat', 'STDOUT', 'irods_policy_storage_tiering')
+
+                    # Ensure that the replica was tiered out to ufs2 and preserved on ufs1, and no replica exists on
+                    # ufs0.
+                    delay_assert_icommand(admin_session, 'ils -L ' + filename, 'STDOUT_SINGLELINE', 'ufs2')
+                    self.assertTrue(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs1'))
+                    self.assertFalse(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs0'))
+
+                    # Now for the actual test. Get the data object in order to trigger a restage.
+                    admin_session.assert_icommand(['iget', '-R', 'ufs2', filename, '-'], 'STDOUT', 'TESTFILE')
+
+                    # Ensure that the restage has been scheduled.
+                    admin_session.assert_icommand('iqstat', 'STDOUT', 'irods_policy_data_movement')
+
+                    # Ensure that the replica is restaged to the minimum restage tier (ufs0) and the replicas on ufs1
+                    # and ufs2 are preserved.
+                    delay_assert_icommand(admin_session, 'ils -L ' + filename, 'STDOUT_SINGLELINE', 'ufs0')
+                    self.assertTrue(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs1'))
+                    self.assertTrue(lib.replica_exists_on_resource(admin_session, logical_path, 'ufs2'))
+
+                    # TODO: #235 - Tiering back out at this point FAILS. Uncomment this section once #235 is resolved.
+                    ## Now, tier out the replica from ufs0 to ufs1 again.
```

Comment:
Are you planning on addressing #235 now that we've reached a decision?

Reply:
Yes, I think it should be relatively trivial to address it, so I will go ahead and do that. Should delete the "comment out" commit as a result. I'll move this PR into draft until that's done.

Reply:
Seems fixed. Test passes now.